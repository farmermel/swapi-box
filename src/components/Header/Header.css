.header {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  margin: 0 0 40px 0; }

.page-header {
  display: flex;
  justify-content: center; }

nav {
  display: flex;
  justify-content: center;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid black;
  background-color: #87bcff; }

nav button {
  border: none;
  margin: 15px;
  background-color: #87bcff;
  border: none; }
  nav button:hover {
    background-color: #DA5344;
    border-radius: 11px;
    opacity: 0.7; }

.active button {
  border-radius: 11px;
  background-color: #DA5344; }

.page-title {
  display: inline-block;
  color: yellow;
  font-family: 'legothickregular', 'Luckiest Guy', cursive;
  font-size: 5em; }

.logo {
  position: absolute;
  left: -9px;
  top: 36px;
  margin: 0; }

.header-logo {
  margin-top: 39px;
  height: 108px;
  width: 145px; }

.people {
  background-size: cover;
  width: 86px;
  height: 70px;
  background-image: url("../../assets/lego-people.png"); }

.planets {
  background-size: cover;
  width: 70px;
  height: 70px;
  background-image: url("../../assets/lego-death-star.png"); }

.vehicles {
  background-size: cover;
  width: 110px;
  height: 70px;
  background-image: url("../../assets/lego-ship.png"); }

.favorites {
  background-size: cover;
  width: 70px;
  height: 70px;
  background-image: url("../../assets/lego-heart.png"); }
