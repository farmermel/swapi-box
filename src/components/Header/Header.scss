.page-title {
  display: inline-block;
  font-family: 'legothickregular';
}

.header-logo {
  width: 70px;
}