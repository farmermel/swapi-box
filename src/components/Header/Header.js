import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import lego from '../../assets/lego.svg';
import yoda from '../../assets/yoda.png';
import './Header.css';

const Header = ({ getPlanets, getVehicles, translate, translated }) => {
  const buttonDisplay = translated 
    ? 'Aoawhraaowo English' 
    : 'Translate to Shyriiwook';

  return (
    <div className='header'>
      <button className='wookiee'
        onClick={() => translate()}>{ buttonDisplay }</button>
      <img src={ lego }
        alt='lego brand'
        className='header-logo'
        id='logo' />
      <div className='page-header'>   
        <h1 className='page-title'>STAR WARS</h1>
        <img src={ yoda } 
          alt='lego yoda'
          className='header-logo' />
      </div>
      <nav>
        <NavLink to={ {pathname: '/people'} }>
          <button className='people'>
          </button>
        </NavLink>
        <NavLink to={ {pathname: '/planets'} }
          onClick={ getPlanets }>
          <button className='planets'>
          </button>
        </NavLink>
        <NavLink to={ {pathname: '/vehicles'} }
          onClick={ getVehicles }>
          <button className='vehicles'>
          </button>
        </NavLink>
        <NavLink to={ {pathname: '/favorites'} }>
          <button className='favorites'>
          </button>
        </NavLink>
      </nav>
    </div>
  );
};

Header.propTypes = {
  getPlanets: PropTypes.func.isRequired,
  getVehicles: PropTypes.func.isRequired,
  translate: PropTypes.func,
  translated: PropTypes.bool
};

export default Header;