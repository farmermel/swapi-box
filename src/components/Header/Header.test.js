import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  it('matches the snapshot', () => {
    let wrapper = shallow(<Header getPlanets={jest.fn()}
                                  getVehicles={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  })
})