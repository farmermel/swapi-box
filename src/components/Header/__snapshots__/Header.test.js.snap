// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    getPlanets={[Function]}
    getVehicles={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="lego brand"
          className="header-logo logo"
          src="lego.svg"
/>,
        <div
          className="page-header"
>
          <h1
                    className="page-title"
          >
                    STAR WARS
          </h1>
          <img
                    alt="lego yoda"
                    className="header-logo"
                    src="yoda.png"
          />
</div>,
        <nav>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to={
                              Object {
                                        "pathname": "/people",
                                      }
                    }
          >
                    <button
                              className="people"
                    />
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    onClick={[Function]}
                    to={
                              Object {
                                        "pathname": "/planets",
                                      }
                    }
          >
                    <button
                              className="planets"
                    />
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    onClick={[Function]}
                    to={
                              Object {
                                        "pathname": "/vehicles",
                                      }
                    }
          >
                    <button
                              className="vehicles"
                    />
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to={
                              Object {
                                        "pathname": "/favorites",
                                      }
                    }
          >
                    <button
                              className="favorites"
                    />
          </NavLink>
</nav>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "lego brand",
          "className": "header-logo logo",
          "src": "lego.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-title"
>
              STAR WARS
</h1>,
            <img
              alt="lego yoda"
              className="header-logo"
              src="yoda.png"
/>,
          ],
          "className": "page-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "STAR WARS",
              "className": "page-title",
            },
            "ref": null,
            "rendered": "STAR WARS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "lego yoda",
              "className": "header-logo",
              "src": "yoda.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to={
                            Object {
                                          "pathname": "/people",
                                        }
              }
>
              <button
                            className="people"
              />
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              onClick={[Function]}
              to={
                            Object {
                                          "pathname": "/planets",
                                        }
              }
>
              <button
                            className="planets"
              />
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              onClick={[Function]}
              to={
                            Object {
                                          "pathname": "/vehicles",
                                        }
              }
>
              <button
                            className="vehicles"
              />
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to={
                            Object {
                                          "pathname": "/favorites",
                                        }
              }
>
              <button
                            className="favorites"
              />
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="people"
/>,
              "to": Object {
                "pathname": "/people",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "people",
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="planets"
/>,
              "onClick": [Function],
              "to": Object {
                "pathname": "/planets",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "planets",
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="vehicles"
/>,
              "onClick": [Function],
              "to": Object {
                "pathname": "/vehicles",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "vehicles",
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="favorites"
/>,
              "to": Object {
                "pathname": "/favorites",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "favorites",
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="lego brand"
            className="header-logo logo"
            src="lego.svg"
/>,
          <div
            className="page-header"
>
            <h1
                        className="page-title"
            >
                        STAR WARS
            </h1>
            <img
                        alt="lego yoda"
                        className="header-logo"
                        src="yoda.png"
            />
</div>,
          <nav>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to={
                                    Object {
                                                "pathname": "/people",
                                              }
                        }
            >
                        <button
                                    className="people"
                        />
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        onClick={[Function]}
                        to={
                                    Object {
                                                "pathname": "/planets",
                                              }
                        }
            >
                        <button
                                    className="planets"
                        />
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        onClick={[Function]}
                        to={
                                    Object {
                                                "pathname": "/vehicles",
                                              }
                        }
            >
                        <button
                                    className="vehicles"
                        />
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to={
                                    Object {
                                                "pathname": "/favorites",
                                              }
                        }
            >
                        <button
                                    className="favorites"
                        />
            </NavLink>
</nav>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "lego brand",
            "className": "header-logo logo",
            "src": "lego.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="page-title"
>
                STAR WARS
</h1>,
              <img
                alt="lego yoda"
                className="header-logo"
                src="yoda.png"
/>,
            ],
            "className": "page-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "STAR WARS",
                "className": "page-title",
              },
              "ref": null,
              "rendered": "STAR WARS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "lego yoda",
                "className": "header-logo",
                "src": "yoda.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to={
                                Object {
                                                "pathname": "/people",
                                              }
                }
>
                <button
                                className="people"
                />
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                onClick={[Function]}
                to={
                                Object {
                                                "pathname": "/planets",
                                              }
                }
>
                <button
                                className="planets"
                />
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                onClick={[Function]}
                to={
                                Object {
                                                "pathname": "/vehicles",
                                              }
                }
>
                <button
                                className="vehicles"
                />
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to={
                                Object {
                                                "pathname": "/favorites",
                                              }
                }
>
                <button
                                className="favorites"
                />
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="people"
/>,
                "to": Object {
                  "pathname": "/people",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "people",
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="planets"
/>,
                "onClick": [Function],
                "to": Object {
                  "pathname": "/planets",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "planets",
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="vehicles"
/>,
                "onClick": [Function],
                "to": Object {
                  "pathname": "/vehicles",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "vehicles",
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="favorites"
/>,
                "to": Object {
                  "pathname": "/favorites",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "favorites",
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
